### ISSUE TITLE
Remove curly "Unnecessary { after if/for-of" autofixes from on-save formatter

### ISSUE THREAD

**Issue by Dimava** [antfu/eslint-config#449]
### Clear and concise description of the problem

Pretty often I find myself writing
```ts
if (foo) {
   bar()
   void 0
}
```
to avoid ESLint collapsing that to
```ts
if (foo)
  bar()
```
when I save the file

### Suggested solution

Move "Unnecessary { after if/for-of" autofixes to on-explicit-format (or on-command-line) formatting so it doesn't autofix while I'm in the middle of writing it

### Alternative

Autosaving may sometimes bypass linting, but I prefer my files not to autosave ever in some cases

### Validations

- [X] Follow our [Code of Conduct](https://github.com/antfu/.github/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guide](https://github.com/antfu/contribute).
- [X] Check that there isn't already an issue that request the same feature to avoid creating a duplicate.

**Comment by nowo** [comment]
You can add `'curly' : [ 'error' , 'multi-line' , 'consistent' ] ,//uniform braces` to the eslint configuration file,or configure your own rules https://eslint.org/docs/latest/rules/curly

**Comment by zanfee** [comment]
Curly rules are actually the only rules we have adapted for our ESLint config at Bechtle. I'm interested in whether there are any objective advantages to using the short form or if this is a personal preference, as Anthony usually has a good reason for these rules.

Our custom rules are as follows:

```js
rules: {
  'curly': ['error', 'all'],
  'style/brace-style': ['error', 'stroustrup'],
},
```

And these are the current rules of @antfu/eslint-config (`allowSingleLine` of the 'style/brace-style' rule is part of [eslint-stylistic defaults](https://github.com/eslint-stylistic/eslint-stylistic/blob/1d3d47ca23ed724210b57440d34b3c3df14079c0/packages/eslint-plugin/configs/customize.ts#L108))

```js
rules: {
  'curly': ['error', 'multi-or-nest', 'consistent'],
  'style/brace-style': ['error', 'stroustrup', { allowSingleLine: true }],
},
```

Here are the docs for [curly](https://eslint.org/docs/latest/rules/curly) and [style/brace-style](https://eslint.style/rules/js/brace-style) rules.

Basically, our rules ensure that there are always curly braces, even for single-line blocks. In my opinion, this has the following advantages:

- No need to manually add curly braces when adding a second line to the block
- Minimal git diff when adding a second line
- Blocks are easier to identify, less room for error

One drawback of our rules is that the code takes up more space. But you can still write blocks without braces and they will be added automatically.

**Comment by ketao4869** [comment]
<details>
<summary>
my eslint.config.js
</summary>

```js
import antfu from '@antfu/eslint-config'  // 3.8.0

export default antfu(
  {
    type: 'app',
    lessOpinionated: false,

    stylistic: {
      indent: 2,
      quotes: 'single',
    },

    typescript: true,
    vue: true,

    // jsonc: false,
    // yaml: false,

    unocss: true,

    ignores: [
      '**/fixtures',
    ],

    formatters: {
      css: true,
      html: true,
    },

  },
  {
    files: ['**/*.vue'],
    rules: {},
  },
  {
    rules: {
      'no-console': 'off',
    },
  },
)
```
</details>

![image](https://github.com/user-attachments/assets/bfe12561-1762-4bf3-ab52-b9b85028752f)

Why do both styles exist? Shouldn't they be formatted as one or the other?
I want to omit brackets for single-line blocks. Do I have to add a configuration? Isn't it the default configuration?

**Comment by antfu** [comment]
The config already changed to accept both style - the original issue should be resolved

**Comment by ketao4869** [comment]
> 配置已经更改为接受两种风格 - 原始问题应该得到解决

如果只想要生效一种，是添加新规则，还是有配置项？
